pkg/api/validation/validation.go

```diff
@@ -762,6 +762,17 @@ func validateHTTPGetAction(http *api.HTTPGetAction) errs.ValidationErrorList {
 	} else if http.Port.Kind == util.IntstrString && len(http.Port.StrVal) == 0 {
 		allErrors = append(allErrors, errs.NewFieldRequired("port"))
 	}
+	supportedSchemes := []api.URIScheme{api.URISchemeHTTP, api.URISchemeHTTPS}
+	unsupportedErrorMessage := fmt.Sprintf("must be a supported scheme for probing: %v", supportedSchemes)
+	validScheme := false
+	for _, scheme := range supportedSchemes {
+		if http.Scheme == scheme {
+			validScheme = true
+		}
+	}
+	if !validScheme {
+		allErrors = append(allErrors, errs.NewFieldInvalid("scheme", http.Scheme, unsupportedErrorMessage))
+	}
```

Comment:
this works, but it might be cleaner to follow this pattern:
```
var supportedGetActionSchemes = util.NewStringSet(string(api.URISchemeHTTP), string(api.URISchemeHTTPS))
...
if !supportedGetActionSchemes.Has(string(http.Scheme)) {
  allErrors = append(allErrors, errs.NewFieldInvalid("scheme", http.Scheme, fmt.Sprintf("must be one of %v", supportedGetActionSchemes.List())))
}
```